seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/source/ClickhouseSourceReader.java

```diff
@@ -97,6 +101,12 @@ record -> {
             }
             this.readerContext.signalNoMoreElement();
             this.splits.clear();
+        } else if (noMoreSplit
+                && splits.isEmpty()
+                && Boundedness.BOUNDED.equals(readerContext.getBoundedness())) {
+            log.info("Closed the bounded ClickHouse source");
+            this.readerContext.signalNoMoreElement();
+            this.splits.clear();
         }
```

Comment:
```suggestion
            this.splits.clear();
        }
        if (noMoreSplit
                && splits.isEmpty()
                && Boundedness.BOUNDED.equals(readerContext.getBoundedness())) {
            log.info("Closed the bounded ClickHouse source");
            this.readerContext.signalNoMoreElement();
            this.splits.clear();
        }
```

Reply:
`Boundedness.BOUNDED.equals(readerContext.getBoundedness())`
This can be removed I think because clickhouse only supports batch mode reading
cc @Hisoka-X

Reply:
Keep it for now. The purpose of keeping it is to unify the processing logic of most connectors, to facilitate reference for other developers and reduce the chance of bugs.

Reply:
> Keep it for now. The purpose of keeping it is to unify the processing logic of most connectors, to facilitate reference for other developers and reduce the chance of bugs.

What if the user uses processing mode to write STREAMING? It's unnecessary code and I think it's better to get rid of it Developers adding features are certainly familiar with their read patterns

Reply:
> What if the user uses processing mode to write STREAMING?

Sorry, I don get it. Could you share more details?

Reply:
> > What if the user uses processing mode to write STREAMING?
> 
> Sorry, I don get it. Could you share more details? You meaning set `job.mode` to `STREAMING`?
```

env {
  job.mode = "STREAMING"
  parallelism = 2
}
source {
  # This is a example source plugin **only for test and demonstrate the feature source plugin**
  Clickhouse {
    host = "127.0.0.1:8123"
    database = "default"
    sql = "select * from students"
    username = "default"
    password = ""
  }
  # If you would like to get more information about how to configure seatunnel and see full list of source plugins,
  # please go to https://seatunnel.apache.org/docs/connector-v2/source/ClickhouseSource
}


sink {
  Console {
  }
  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
  # please go to https://seatunnel.apache.org/docs/connector-v2/sink
}

```

So for example in this case the user will never exit the program, essentially no matter how you configure the read mode clickhouse only supports batch mode, right

Reply:
If users set to STREAMING, but source only supported BATCH mode. The job will execute as BATCH mode. It depends on https://github.com/apache/seatunnel/blob/a8d0d4ce77a1ac9116f93381dfe296d22d8f0e8b/seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/source/ClickhouseSource.java#L160

Reply:
Understood, but an extra judgment is a redundant operation